pkg/csi_driver/controller.go

```diff
@@ -214,7 +214,14 @@ func (s *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVolu
 		}
 		if createErr != nil {
 			glog.Errorf("Create volume for volume Id %s failed: %v", volumeID, createErr)
-			return nil, status.Error(codes.Internal, createErr.Error())
+			// Return InvalidArgument code for googleapi errors that are caused by users such as Error 429: Quota limit exceeded
+			// and Error 403: Permission Denied.
+			if file.IsUserError(createErr) {
```

Comment:
I wonder if it would be easier to just do return nil,  FilestoreAPIError(createErr), and then have FilestoreApiError map filestore API error codes to grpc error codes. That way if it ever gets more complicated we have all the error logic in the same place.

@saikat-royc ?

Reply:
Right now that seems a little difficult for the IsNotFoundErr function. (I am assuming we would want to do this for both the IsNotFoundErr and the IsUserErr functions). There are 6 if checks for  `if file.IsNotFoundErr(err)` and it doesn't always return the same grpc error code. Sometimes it [returns codes.NotFound](https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/master/pkg/csi_driver/controller.go#L144-L147),  sometimes it [doesn't return an error](https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/master/pkg/csi_driver/controller.go#L279-L281), some times it returns codes.Internal. So sometimes in different contexts a googleapi not found error would mean you would want to return a different  grpc error code, or none at all.

Reply:
Good point, makes sense to me.